src/core/apple/AppleService.ts

```diff
@@ -34,15 +34,15 @@ async function getMatchedKeyBy(kid: string, alg: string): Promise<AppleJWK> {
     }
 }
 
-export async function verifyAndDecodeAppleToken(identityToken: string): Promise<AppleUserInfo> {
+export async function verifyAndDecodeAppleToken(identityToken: string, appVersion: string): Promise<AppleUserInfo> {
     const headerOfIdentityToken: JwtHeader = JSON.parse(Buffer.from(identityToken.substr(0, identityToken.indexOf('.')), 'base64').toString());
     const appleJwk: AppleJWK = await getMatchedKeyBy(headerOfIdentityToken.kid, headerOfIdentityToken.alg);
     const publicKey: string = pemjwk.jwk2pem(appleJwk);
     try {
         jwt.verify(identityToken, publicKey, {
             algorithms: [appleJwk.alg],
             issuer: "https://appleid.apple.com",
-            audience: "com.wafflestudio.snutt"
+            audience: appVersion.split(".")[0] === "3" ? "com.wafflestudio.snutt3" : "com.wafflestudio.snutt",
```

Comment:
현재까지 진행+구상된 것은 이렇습니다.

- 계정간 앱 마이그레이션을 하면, prod 앱은 기존 번들ID(`com.wafflestudio.snutt`)를 그대로 사용할 수 있다.
- 테스트플라이트에 prod 앱과 dev 앱을 둘 다 올리려면 dev 앱은 별도의 번들ID(e.g. `com.wafflestudio.snutt.dev`를 가져야 한다.

따라서 appVersion이 아니라 appType에 따라서 아래와 같이 분기처리를 해야 할 것 같습니다. (appVersion을 헤더에 포함하지 않는 구버전에 대한 핸들링도 필요할 것 같습니다.)

```suggestion
            audience: appType === "release" ? "com.wafflestudio.snutt" : "com.wafflestudio.snutt.dev",
```

추후에 변경사항이 생기면 그때는 제가 직접 PR 올리도록 하겠습니다!

Reply:
그럼 이 PR 은 일단 close 할게요?

Reply:
@shp7724 반영해서, commit 하고 PR 수정해서 다시 열었습니다.